src/brucke_health_check.erl

```diff
@@ -0,0 +1,124 @@
+
+init() ->
+  case application:get_env(?APPLICATION, healthcheck) of
+    {ok, true} ->
+      Port = application:get_env(?APPLICATION, healthcheck_port, 8080),
+      Dispatch = cowboy_router:compile([{'_', [{'_', brucke_http_handler, []}]}]),
+      {ok, _} = cowboy:start_clear(http, 1, [{port, Port}], #{env => #{dispatch => Dispatch}}),
+      ok;
+    _ -> ok
+  end.
+
+-spec get_report_data(proplists:proplist()) -> {boolean(), map()}.
+get_report_data(Params) ->
+  AskClients = proplists:get_value(?CLIENTS, Params, true),
+  {AllCliends, Status} = get_all_clients(AskClients, true),
+  AskRoutes = proplists:get_value(?ROUTES, Params, true),
+  {AllRoutes, UStatus} = get_all_routes(AskRoutes, Status),
+  {UStatus, #{?CLIENTS => AllCliends, ?ROUTES => AllRoutes}}.
+
+
+%% @private
+-spec get_all_clients(boolean(), boolean()) -> {list(map()), boolean()}.
+get_all_clients(true, Status) ->
+  AllClients = brucke_config:all_clients(),
+  lists:foldl(
+    fun({ClientId, _, _}, {Acc, AccStatus}) ->
+      case whereis(ClientId) of
+        undefined -> {[#{ClientId => undefined} | Acc], false};
+        _ -> {[#{ClientId => ok} | Acc], AccStatus}
+      end
+    end, {[], Status}, AllClients);
+get_all_clients(_, Status) -> {[], Status}.
+
+%% @private
+-spec get_all_routes(boolean(), boolean()) -> {list(map()), boolean()}.
+get_all_routes(true, Status) ->
+  AllRoutes = brucke_config:all_routes(),
+  AllRouteSups = brucke_sup:get_children(),
+  DiscardedRoutes = brucke_routes:get_by_status(skipped),
+  {Working, Status1} = get_working_routes(AllRoutes, AllRouteSups, Status),
+  apply_discarded(Working, DiscardedRoutes, Status1);
+get_all_routes(_, Status) -> {[], Status}.
+
+%% @private
+get_working_routes(AllRoutes, AllRouteSups, Status) ->
+  lists:foldl(
+    fun(#route{upstream = Upstream, downstream = Downstream}, {Acc, UStatus}) ->
+      {Route, Status1} = check_route(Upstream, AllRouteSups, UStatus),
+      {UpstreamId, UpstreamTopic} = Upstream,
+      {DownstreamId, DownstreamTopic} = Downstream,
+      Acc1 = [Route#{?UPSTREAM => #{?CLIENT_ID => UpstreamId, ?TOPIC => UpstreamTopic},
+        ?DOWNSTREAM => #{?CLIENT_ID => DownstreamId, ?TOPIC => DownstreamTopic}} | Acc],
+      {Acc1, Status1}
+    end, {[], Status}, AllRoutes).
+
+%% @private
+apply_discarded(Working, DiscardedRoutes, Status) ->
+  lists:foldl(  % at least one discarded route is enough to toggle status to false.
+    fun(#route{upstream = {Id, Topic}, reason = Reason}, {Acc, _}) ->
+      {[#{?UPSTREAM => #{?CLIENT_ID => Id, ?TOPIC => Topic}, ?STATUS => Reason} | Acc], false}
+    end, {Working, Status}, DiscardedRoutes).
+
+%% @private
+check_route(Upstream, Routes, Status) ->
+  case lists:keyfind(Upstream, 1, Routes) of
+    {_, RoutePid, _, _} when is_pid(RoutePid) ->  % ok / dead (but pid found in sup tree)
+      check_process(RoutePid, Status);
+    {_, Status, _, _} ->  % undefined / restarting
+      {#{?STATUS => Status}, false};
```

Comment:
here and line 101, add `members => []`
maybe add a function `make_route_satus(Status, Members)` to ensure unified schema in JSON response.